Daf.Meta.Editor/ViewModels/DataSourceViewModel.cs

```diff
@@ -0,0 +1,247 @@
+			}
+		}
+
+		[Category("General")]
+		public bool? ContainsMultiStructuredJson
+		{
+			get => DataSource.ContainsMultiStructuredJson;
+			set
+			{
+				SetProperty(DataSource.ContainsMultiStructuredJson, value, DataSource, (dataSource, containsMultiStructuredJson) => dataSource.ContainsMultiStructuredJson = containsMultiStructuredJson, true);
+			}
+		}
+
+		[Category("General")]
+		public string? FileName
+		{
+			get => DataSource.FileName;
+			set
+			{
+				SetProperty(DataSource.FileName, value, DataSource, (dataSource, fileName) => dataSource.FileName = fileName, true);
+			}
+		}
+
+		[Category("General")]
+		public string? IncrementalStagingColumn
+		{
+			get => DataSource.IncrementalStagingColumn;
+			set
+			{
+				SetProperty(DataSource.IncrementalStagingColumn, value, DataSource, (dataSource, incrementalStagingColumn) => dataSource.IncrementalStagingColumn = incrementalStagingColumn, true);
+			}
+		}
+
+		[DataType(DataType.MultilineText)]
+		[Category("General")]
+		public string? IncrementalQuery
+		{
+			get => DataSource.IncrementalQuery;
+			set
+			{
+				SetProperty(DataSource.IncrementalQuery, value, DataSource, (dataSource, incrementalQuery) => dataSource.IncrementalQuery = incrementalQuery, true);
+			}
+		}
+
+		[Category("General")]
+		public string? BusinessDateColumn
+		{
+			get => DataSource.BusinessDateColumn;
+			set
+			{
+				SetProperty(DataSource.BusinessDateColumn, value, DataSource, (dataSource, businessDateColumn) => dataSource.BusinessDateColumn = businessDateColumn, true);
+			}
+		}
+
+		[Category("General")]
+		[Description("The custom select query that is run against the load table when loading the staging table.")]
+		public string? SqlSelectQuery
+		{
+			get => DataSource.SqlSelectQuery;
+			set
+			{
+				SetProperty(DataSource.SqlSelectQuery, value, DataSource, (dataSource, sqlSelectQuery) => dataSource.SqlSelectQuery = sqlSelectQuery, true);
+			}
+		}
+
+		[Category("Azure")]
+		public string? AzureLinkedServiceReference
+		{
+			get => DataSource.AzureLinkedServiceReference;
+			set
+			{
+				SetProperty(DataSource.AzureLinkedServiceReference, value, DataSource, (dataSource, azureLinkedServiceReference) => dataSource.AzureLinkedServiceReference = azureLinkedServiceReference, true);
+			}
+		}
+
+		[Category("General")]
+		[SelectorStyle(SelectorStyle.ComboBox)]
+		public Build Build
+		{
+			get => DataSource.Build;
+			set
+			{
+				SetProperty(DataSource.Build, value, DataSource, (dataSource, build) => dataSource.Build = build, true);
+			}
+		}
+
+		[Category("General")]
+		public string? ErrorHandling
+		{
+			get => DataSource.ErrorHandling;
+			set
+			{
+				SetProperty(DataSource.ErrorHandling, value, DataSource, (dataSource, errorHandling) => dataSource.ErrorHandling = errorHandling, true);
+			}
+		}
+
+		// TODO: This is necessary but should be a property on SatellitesViewModel instead.
```

Comment:
Should this be fixed?

Reply:
Unsure, but leaning towards no. SatellitesViewModel needs to access that property on DataSource, but it seems reasonable to avoid any other view model referring directly to DatasourceViewModel.DataSource unless necessary.

As we've discussed before, DataSourceViewModel currently works as a VM for the General tab, as well as making available some properties for other view models that needs them. So it's performing sort of an awkward role. But considering the structure of the view models as they are now it may be reasonable to keep it the way it is.

Reply:
The same question needs to be posed for all properties on DataSourceViewModel that has the Browsable False attribute. But I think we've discussed this and concluded that the current implementation is good enough for now and can be tweaked later.